ddtrace/contrib/urllib3/patch.py

```diff
@@ -0,0 +1,150 @@
+from ...utils.formats import asbool
+from ...utils.formats import get_env
+from ...utils.wrappers import unwrap as _u
+
+
+# Ports which, if set, will not be used in hostnames/service names
+DROP_PORTS = (80, 443)
+
+# Initialize the default config vars
+config._add(
+    "urllib3",
+    {
+        "_default_service": "urllib3",
+        "distributed_tracing": asbool(get_env("urllib3", "distributed_tracing", default=True)),
+        "analytics_enabled": asbool(get_env("urllib3", "analytics_enabled", default=False)),
+        "analytics_sample_rate": float(get_env("urllib3", "analytics_sample_rate", default=1.0)),
+        "trace_query_string": asbool(get_env("urllib3", "trace_query_string", default=False)),
+        "split_by_domain": asbool(get_env("urllib3", "split_by_domain", default=False)),
+    },
+)
+
+
+def patch():
+    """Enable tracing for all urllib3 requests"""
+    if getattr(urllib3, "__datadog_patch", False):
+        return
+    setattr(urllib3, "__datadog_patch", True)
+
+    _w("urllib3", "connectionpool.HTTPConnectionPool.urlopen", _wrap_urlopen)
+
+
+def unpatch():
+    """Disable trace for all urllib3 requests"""
+    if getattr(urllib3, "__datadog_patch", False):
+        setattr(urllib3, "__datadog_patch", False)
+
+        _u(urllib3.connectionpool.HTTPConnectionPool, "urlopen")
+
+
+def _infer_argument_value(args, kwargs, pos, kw, default=None):
+    """
+    This function parses the value of a target function argument that may have been
+    passed in as a positional argument or a keyword argument. Because monkey-patched
+    functions do not define the same signature as their target function, the value of
+    arguments must be inferred from the packed args and kwargs.
+
+    Keyword arguments are prioritized, followed by the positional argument, followed
+    by the default value, if any is set.
+
+    :param args: Positional arguments
+    :param kwargs: Keyword arguments
+    :param pos: The positional index of the argument if passed in as a positional arg
+    :param kw: The name of the keyword if passed in as a keyword argument
+    :param default: The default value to return if the argument was not found in args or kwaergs
+    :return: The value of the target argument
+    """
+    if kw in kwargs:
+        return kwargs[kw]
+
+    if pos < len(args):
+        return args[pos]
+
+    return default
+
+
+def _wrap_urlopen(func, instance, args, kwargs):
+    """
+    Wrapper function for the lower-level urlopen in urllib3
+
+    :param func: The original target function "urlopen"
+    :param instance: The patched instance of ``HTTPConnectionPool``
+    :param args: Positional arguments from the target function
+    :param kwargs: Keyword arguments from the target function
+    :return: The ``HTTPResponse`` from the target function
+    """
+    request_method = _infer_argument_value(args, kwargs, 0, "method")
+    request_url = _infer_argument_value(args, kwargs, 1, "url")
```

Comment:
These arguments are not also kwargs: 

https://github.com/urllib3/urllib3/blob/d0b20763f55536aec43caae9d180aa16c7b77d09/src/urllib3/connectionpool.py#L473-L474

```suggestion
    request_method = args[0]
    request_url = args[1]
```